### DISCUSSION TITLE
Tipos de empresa para trabalho Front-end

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#422] [Discussão]
Opa pessoal beleza? Queria abrir um tópico relacionado mais as formas de trabalho, onde o pessoal desde novatos até seniors com mais de 10 anos de profissão relatem suas experiências e prós e contras dos tipos de empresas que encontramos no mercado.
Acho que existem alguns: Startups de ínicio com pouco capital, Startups com bastante capital,Agências de web,Agências de Publicidade, Empresas de Tecnologia,Empresas de Consultoria,Empresas de E-Commerce,Freelancing com pequenos negócios,Freelancing com agências,Freelancing p / exterior , não sei todos os tipos, esses foram o que vieram a minha cabeça,queria ver o pró e os contras de cada uma, qual gostam mais, qual buscam no momento, e os que conseguiram se sentem satisfeitos com isso. Valeu.

**Comment by Cleysonlb** [comment]
Bom eu comecei na área a 4 anos atrás, meu primeiro emprego como front end foi em uma agencia (entrei sem nem saber oq era front end), foi uma experiência mediana, serviu só introduzir na área e entender um pouco como as coisas funcionavam.
Moro em uma cidade pequena, sem muito mercado por isso entrei no mundo freelance/remoto, passei por outras agencias trabalhando home office oq tbm não me fez evoluir muito.... Bom, mas sempre me mantive estudando coisas por fora e tentando entender outras realidades.
Hoje trabalho em uma startup, trabalho inteiramente em produto e está sendo uma ótima experiência, estou podendo explorar melhor oq sei e melhorar em outros pontos, é um desafio, é corrido mas é FODA!

Pró de agencia: Você lida com projetos dos mais diversos nichos de mercado, projetos complexos (layout) e desafiadores, você acaba aprendendo a codar mais rapido pq o ritmo é tenso (não necessariamente codar melhor)

Contra: se vc nao tem perfil de estudar por fora, vc vai ficar estagnado em cortar PSD e liberar layouts.

Hoje to me bandeando para o lado UX da coisa, querendo entrar nesse mercado, aplico alguns conceitos aqui na empresa onde trabalho e é bem legal!

**Comment by laurindo** [comment]
Meus últimos dois empregos foram focados em produto. Que pra mim é algo
muito desafiador, divertido e empolgante de se trabalhar  pois sempre tem
desafios novos, um novo problema pra resolver, novas tecnologias a
utilizar.
Na maioria das vezes, eu como desenvolvedor sou responsável por ajudar na
decisão de usar ou não determinada tecnologia.
Utilizamos muito de metodologias ágeis para entregar pequenas
funcionalidades.

Já trabalhei em agências e empresas de outsourcing.

Mas trabalhar com produto é mais bacana. Você se torna dono daquilo que vc
faz. E claro você não para de estudar.

**Comment by alexislds** [comment]
Comecei em Julho de 2016 na area como estagiario na area de Front-end, tinha estudado sobre HTML CSS e SASS e um pouco de responsivo, pelo ALURA. Entrando comecei a pegar PSD's e converter para sites, seguindo os padrões da empresa, como estilo de codigo, onde comentar ou não comentar o codigo. Aumentar tamanho de elementos usando padding ou height / width.
Onde usar imagem ou usar um linear-gradient, fui pegando as boas praticas, e após isso comecei a estudar JS puro por conta, para futuramente usar no emprego.

Mas de tecnologias acabamos usando:

HTML/CSS, SASS, Javascript puro, GIT, Gulp. Photoshop e SVN.

**Comment by brunoravanhani** [comment]
A minha experiência com front-end começou numa empresa antiga, porém pequena (que eu ainda trabalho), resumindo a história a empresa tem 30 anos e já foi grande hoje é pequena (uns 15 funcionários). 

Basicamente os pontos positivos é que é uma boa empresa pra se trabalhar, as pessoas bem tranquilas apesar dos prazos apertados às vezes, temos uma certa liberdade aqui.

Os pontos negativos principais é que, pelo menos na minha área não tem nenhum senior, então meio que padrões, boas práticas e aprendizado tem que partir basicamente de mim. E o chefe é chefe, tipo de tribo mesmo, as demandas e principalmente as interfaces vêm dele, as vezes desenhadas em papel, e é ele quem decide.

**Comment by leandroruel** [comment]
> e as interfaces vêm dele, as vezes desenhadas em papel, e é ele quem decide.

@brunoravanhani isso é péssimo, ele é designer?

**Comment by brunoravanhani** [comment]
Não. Isso é péssimo mesmo. Mas eu estou tentando mudar o mindset aqui aos poucos
